pkg/api/cluster_types_test.go

```diff
@@ -157,6 +230,75 @@ func TestGetAvailableAndReadyStrimziVersions(t *testing.T) {
 	}
 }
 
+func Test_Cluster_GetLatestAvailableAndReadyStrimziVersion(t *testing.T) {
+	tests := []struct {
+		name    string
+		cluster func() *Cluster
+		want    *StrimziVersion
+		wantErr bool
+	}{
+		{
+			name: "When cluster has a non empty list of available strimzi versions those ready returned",
+			cluster: func() *Cluster {
+				inputStrimziVersions := []StrimziVersion{
+					{Version: "v3", Ready: true},
+					{Version: "v7", Ready: true},
+					{Version: "v9", Ready: false},
+				}
+				inputStrimziVersionsJSON, err := json.Marshal(inputStrimziVersions)
+				if err != nil {
+					panic(err)
+				}
+				res := Cluster{AvailableStrimziVersions: inputStrimziVersionsJSON}
+				return &res
+			},
+			want:    &StrimziVersion{Version: "v7", Ready: true},
+			wantErr: false,
+		},
+		{
+			name: "When cluster has an empty list of available strimzi nil is returned",
+			cluster: func() *Cluster {
+				inputStrimziVersions := []StrimziVersion{}
+				inputStrimziVersionsJSON, err := json.Marshal(inputStrimziVersions)
+				if err != nil {
+					panic(err)
+				}
+				res := Cluster{AvailableStrimziVersions: inputStrimziVersionsJSON}
+				return &res
+			},
+			want:    nil,
+			wantErr: false,
+		},
+		{
+			name: "When cluster has a nil list of available strimzi nil is returned",
+			cluster: func() *Cluster {
+				res := Cluster{AvailableStrimziVersions: nil}
+				return &res
+			},
+			want:    nil,
+			wantErr: false,
+		},
+		{
+			name: "When cluster has an invalid JSON an error is returned",
+			cluster: func() *Cluster {
+				res := Cluster{AvailableStrimziVersions: []byte(`"keyone": valueone`)}
+				return &res
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
+			res, err := tt.cluster().GetLatestAvailableAndReadyStrimziVersion()
+			g.Expect(err != nil).To(Equal(tt.wantErr))
```

Comment:
Please, add a test to check that the error we received was the error we were expecting

Reply:
I don't think this test needs to check specific error types. We are only interested on whether there's been an error